### ISSUE TITLE
CA1508 (Avoid dead conditional code) - very strange false positive

### ISSUE THREAD

**Issue by kindermannhubert** [dotnet/roslyn-analyzers#5005]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/CA1508): `Avoid dead conditional code`

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 5.0.3 (Latest)

### Describe the bug

False positive in following code:
```c#
static void M(C1 val1, C1 val2)
{
    bool b1 = true;
    bool b2 = true;
    while (true)
    {
        if (val1.A.X == 1 && val2.A.X == 1) //CA1508: 'val2.A.X == 1' is always 'true'
        {
        }
        b1 = GetBool(val1);
        b2 = GetBool(val2);
    }
}
static bool GetBool(C1 x) => true;
class C1 { public C2 A { get; } }
class C2 { public int X { get; } }
```

### Expected behavior
Diagnostic is not reported in this case.

**Comment by mavasani** [comment]
This does not repro on the latest .NET 8 preview SDK
